### PR TITLE
Fix conn.getInterpolateSetting() always return True

### PR DESCRIPTION
**Pull request by will-moore**
This fixes a bug in Blitz Gateway (#4191) where conn.getInterpolateSetting() always returns True.
See https://www.openmicroscopy.org/community/viewtopic.php?f=4&t=7991&p=16657

To test
```
$ bin/omero config set omero.client.viewer.interpolate_pixels false
```
Restart omero server and log out of web client.
Then log in and check that viewer has interpolation off by default.

Question: is there any way we could have tests for config service methods within Blitz Gateway? Seems tricky if they require a server restart?

### CODE REVIEW

**Comment by joshmoore** [comment]
> Question: is there any way we could have tests for config service methods within Blitz Gateway? Seems tricky if they require a server restart?

An admin account should be able to set the config value. We'll have to be slightly careful that there aren't race conditions. It you have a small set of properties so that we could define "flavors" of integration servers, we could periodically swap and use [skipIf](https://github.com/openmicroscopy/openmicroscopy/blob/v5.2.1/components/tools/OmeroPy/test/integration/test_mail.py#L86).


Quick aside: the calls to getConfigValue are seeing 30-second hangs on some servers. Considering the number called, this is quite the overhead.

**Comment by will-moore** [comment]
@joshmoore Simple test of conn.getInterpolateSetting() pushed.
This seems to work OK.
Not sure I understand your reference to skipIf?

**Comment by mtbc** [comment]
```
test/integration/gatewaytest/test_config_service.py::testInterpolateSetting PASSED
```

Works just fine in the viewer respecting the setting of `omero.client.viewer.interpolate_pixels`.

CI being green tomorrow morning, seems to me to be good to merge.

**Comment by joshmoore** [comment]
> Not sure I understand your reference to skipIf?

Tests can be written which expect each of a couple of settings. i.e. you could write a `def testInterpolateOn` and a `def testInterpolateOff` but of which start with a call to `skipIf`. If the server has been set with interpolation on, then only `testInterpolateOn` would run for example. The benefit here is that it doesn't require a code change to test the other configuration, just a (possibly temporary) PR against mgmt_tools.

**Comment by will-moore** [comment]
@joshmoore The approach I've used in this test is OK? Setting the config and testing each value (true / false)?

**Comment by joshmoore** [comment]
It will likely work in the short-term, but there is certainly a race-condition here if any other tests (run in parallel) assume either `interpolate` is on or off. Minimally, it might help to reset the value to the original value in a try/finally block.

We might should discuss. cc @sbesson

**Comment by jburel** [comment]
Maybe double-check insight too since the property is also used in insight
i.e. check that both clients are in synch

**Comment by aleksandra-tarkowska** [comment]
Test is failing https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-python/191/testReport/OmeroPy.test.integration.gatewaytest/test_config_service/testInterpolateSetting/

File: components/tools/OmeroPy/test/integration/gatewaytest/test_config_service.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    """
+    gatewaywrapper.loginAsAdmin()
+    configService = gatewaywrapper.gateway.getConfigService()
+
+    try:
+        configService.setConfigValue(
+            "omero.client.viewer.interpolate_pixels", "false")
+
+        gatewaywrapper.loginAsAuthor()
+        assert gatewaywrapper.gateway.getInterpolateSetting()
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroPy/test/integration/gatewaytest/test_config_service.py]: this code test `True`, you are missing `not`

**Comment by will-moore** [comment]
Thanks @aleksandra-tarkowska - I had it passing locally because my ```target``` BlitzGateway had got out of sync with ```src```. Fixed now.

**Comment by will-moore** [comment]
@jburel I'm not sure if you're asking for anything else in this PR above?

**Comment by jburel** [comment]
@will-moore: it was just a quick check to make sure that both clients have the interpolation on/off depending on the value set in the configuration.

**Comment by will-moore** [comment]
@jburel I tested Insight the other day when I responded to https://www.openmicroscopy.org/community/viewtopic.php?f=4&t=7991&p=16657 and it was working.
The fix was tested in web above. So now that tests are passing (https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-python/194/testReport/OmeroPy.test.integration.gatewaytest/test_config_service/testInterpolateSetting/) I think we are good to merge?

**Comment by jburel** [comment]
@will-moore: thanks 
Merging

File: components/tools/OmeroPy/test/integration/gatewaytest/test_config_service.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        gatewaywrapper.loginAsAuthor()
+        assert not gatewaywrapper.gateway.getInterpolateSetting()
+
+    finally:
+        # try/finally to make sure that we
+        # set back to 'true' (default value) to clean up
+        gatewaywrapper.loginAsAdmin()
+        configService = gatewaywrapper.gateway.getConfigService()
+        configService.setConfigValue(
+            "omero.client.viewer.interpolate_pixels", "true")
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroPy/test/integration/gatewaytest/test_config_service.py]: you should use ``getConfigService().getConfigDefaults()`` to reset to what is default rather then hard code it. IF default change this test will affect others.

Comment (will-moore) [inline components/tools/OmeroPy/test/integration/gatewaytest/test_config_service.py]: See https://github.com/openmicroscopy/openmicroscopy/pull/4461

**Pull request merged**